### PR TITLE
**Requires LOK Version 0.4.1*** update overlays

### PR DESCRIPTION
**Pull request by whutchinson98**
Added in overlay colorizer code into qa/index.js for testing
Added in ability to save a document to an array buffer and load a document from an array buffer

### CODE REVIEW

File: src/electron/office/document_client.cc [hunk 1]
```diff
@@ -399,6 +400,18 @@ v8::Local<v8::Value> DocumentClient::GotoOutline(int idx,
...
 
+v8::Local<v8::ArrayBuffer> DocumentClient::SaveToMemory(gin::Arguments* args) {
+  char* pOutput = nullptr;
+  size_t size_in_bytes = document_->saveToMemory(&pOutput);
+
+  v8::Isolate* isolate = args->isolate();
+  v8::Local<v8::ArrayBuffer> array_buffer =
+        v8::ArrayBuffer::New(isolate, size_in_bytes);
+  auto backing_store = array_buffer->GetBackingStore();
+  std::memcpy(array_buffer->GetBackingStore()->Data(), pOutput, size_in_bytes);
```
Comment (chase) [inline src/electron/office/document_client.cc:411]: After the copy, should `free(pOutput)` to prevent a mem leak

File: qa/testing/example.docx
Comment (chase) [inline qa/testing/example.docx:1]: Not sure if it's GitHub being silly, but when I try downloading this doc and opening it in LibreOffice it says it is corrupted.

**Review by chase — CHANGES_REQUESTED** [review]
Minor things to address, otherwise solid.

File: qa/testing/example.docx
Comment (whutchinson98) [reply qa/testing/example.docx:1]: Oh I think I accidentally uploaded the corrupted file. I'll upload a good copy of it

File: src/electron/default_app/icon.png
[the remark this answers is not in the document]
Comment (whutchinson98) [reply src/electron/default_app/icon.png:1]: This is that stupid CRLF bug thing. I genuinely cannot seem to get rid of it

**Review by chase — APPROVED** [review]

**Pull request merged**